edgedns/client.go

```diff
@@ -0,0 +1,110 @@
+package edgedns
+
+import (
+	"net/http"
+	"net/url"
+
+	"github.com/corbaltcode/go-akamai"
+	"github.com/corbaltcode/go-akamai/internal/request"
+)
+
+const defaultTTL = 300
+
+const (
+	RecordTypeCNAME = "CNAME"
+	RecordTypeTXT   = "TXT"
+)
+
+type TypesResponse struct {
+	Types []string `json:"types"`
+}
+
+type Recordset struct {
+	Name  string   `json:"name"`
+	Type  string   `json:"type"`
+	TTL   int      `json:"ttl"`
+	Rdata []string `json:"rdata"`
+}
+
+type RecordsetQueryArgs struct {
+	Page     int
+	PageSize int
+	Search   string
+	ShowAll  bool
+	SortBy   string
+	Types    string
+}
+
+type RecordsetResponse struct {
+	Metadata   MetadataH   `json:"metadata"`
+	Recordsets []Recordset `json:"recordsets"`
+}
+
+type MetadataH struct {
+	LastPage      int  `json:"lastPage"`
+	Page          int  `json:"page"`
+	PageSize      int  `json:"pageSize"`
+	ShowAll       bool `json:"showAll"`
+	TotalElements int  `json:"totalElements"`
+}
+
+type Client struct {
+	Credentials akamai.Credentials
+}
+
+func handleQueryArgs(q url.Values) string {
+	s := q.Encode()
+	if len(s) == 0 {
+		return s
+	}
+	return "?" + s
+}
+
+func (c *Client) ListRecordsets(zone string, queryArgs url.Values) (*RecordsetResponse, error) {
+	var rs RecordsetResponse
+	err := request.DoJSON(c.Credentials, http.MethodGet, "/config-dns/v2/zones/"+zone+"/recordsets"+handleQueryArgs(queryArgs), nil, &rs)
+	if err != nil {
+		return nil, err
+	}
+	return &rs, nil
+}
+
+func (c *Client) RetrieveRecordsetTypes(zone, name string) (*TypesResponse, error) {
```

Comment:
IMO, the verb `Get` is better because it's shorter, more Go-like, and more consistent with the other packages in the module.